CliWrap.Tests/ExitConditionSpecs.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Xunit;
+
+namespace CliWrap.Tests;
+
+public class ExitConditionSpecs()
+{
+    [Fact(Timeout = 15000)]
+    public async Task I_can_execute_a_command_that_creates_child_process_resuing_standard_output_and_finish_after_child_process_exits()
+    {
+        // Arrange
+        var cmd = this.PrepareCommand(line => { });
+
+        // Act
+        var executionStart = DateTime.UtcNow;
+        var result = await cmd.ExecuteAsync();
+        var executionFinish = DateTime.UtcNow;
+
+        // Assert
+        executionFinish
+            .Subtract(executionStart)
+            .Should()
+            .BeGreaterThanOrEqualTo(TimeSpan.FromSeconds(3));
```

Comment:
Probably a more reliable way to test this would be to read the stdout. The `sleep` command prints `Done.` after it finishes, so you can use that to detect whether the grandchild process has finished or not.

Reply:
Unfortunately it's not possible. The main process is finished almost instantly, and there is nothing we can do to catch the output after that. We can handle the output and wait for for child process to be finished, but then it breaks the whole point of doing this test, where we are interested in specific scenario when parent triggers child and then dies instantly.

Reply:
I meant to test the current behavior (`PipesClosed`), wouldn't `Done` appear in the output since the command would only complete after the grandchild process exits?

Reply:
Only if we make main process to wait for the child. In the `CliWrap.Tests.Dummy` I implemented `run process` command that doesn't wait for the child, because this is the behavior that can't be handled currrently.

Reply:
In general it seems difficult to create a single test that works reliably on all environments including .NET framework and Linux. In Linux, we should somehow use `nohup` and background processing with bash to reproduce the issue, because the dummy app can't reproduce it.
On .NET Framework those tests also fail, but I don't know why yet.

Reply:
I thought the issue is that, regardless of whether the child process waits for the grandchild or not, because the output streams are inherited and CliWrap waits for them to close, the command effectively finishes when all grandchildren exit.